usd/src/usd_parser/USDWorld.cc

```diff
@@ -94,6 +103,35 @@ namespace usd
       std::vector<std::string> primPathTokens =
         ignition::common::split(primPath, "/");
 
+      // This assumption on the scene graph wouldn't hold if the usd does
+      // not come from Isaac Sim
+      if (primPathTokens.size() == 1 && !prim.IsA<pxr::UsdGeomCamera>()
+          && !prim.IsA<pxr::UsdGeomScope>()
+          && !prim.IsA<pxr::UsdPhysicsScene>()
+          && !prim.IsA<pxr::UsdLuxBoundableLightBase>()
+          && !prim.IsA<pxr::UsdLuxNonboundableLightBase>())
+      {
+        sdf::Model model = sdf::Model();
+        model.SetName(primPathTokens[0]);
+        _world.AddModel(model);
+        modelPtr = _world.ModelByName(primPathTokens[0]);
+
+        ignition::math::Pose3d pose;
+        ignition::math::Vector3d scale{1, 1, 1};
+
+        GetTransform(
+          prim,
+          usdData,
+          pose,
+          scale,
+          model.Name());
+        modelPtr->SetRawPose(pose);
+
+        modelPtr->SetStatic(!prim.HasAPI<pxr::UsdPhysicsRigidBodyAPI>());
+
+        _world.AddModel(model);
```

Comment:
I don't think there's any need for `modelPtr` in this `if` block. We should be able to just use the `sdf::Model` object, so that the code would look something like this (note how the `sdf::Model` object is added to the world after all of its attributes are set):
```suggestion
        sdf::Model model = sdf::Model();
        model.SetName(primPathTokens[0]);

        ignition::math::Pose3d pose;
        ignition::math::Vector3d scale{1, 1, 1};
        GetTransform(
          prim,
          usdData,
          pose,
          scale,
          model.Name());
        model.SetRawPose(pose);

        model.SetStatic(!prim.HasAPI<pxr::UsdPhysicsRigidBodyAPI>());

        _world.AddModel(model);
```